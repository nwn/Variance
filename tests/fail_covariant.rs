mod common;

use common::*;

fn covariant_fail<'a>() {
    let _co: Co<Lifetime<'static>> = Co(
        Covariant::<Lifetime<'a>>::default(),
    );
}

fn main() {}
