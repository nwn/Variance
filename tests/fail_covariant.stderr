error[E0495]: cannot infer an appropriate lifetime due to conflicting requirements
 --> $DIR/fail_covariant.rs:6:38
  |
6 |       let _co: Co<Lifetime<'static>> = Co(
  |  ______________________________________^
7 | |         Covariant::<Lifetime<'a>>::default(),
8 | |     );
  | |_____^
  |
note: first, the lifetime cannot outlive the lifetime `'a` as defined on the function body at 5:19...
 --> $DIR/fail_covariant.rs:5:19
  |
5 | fn covariant_fail<'a>() {
  |                   ^^
note: ...so that the expression is assignable
 --> $DIR/fail_covariant.rs:7:9
  |
7 |         Covariant::<Lifetime<'a>>::default(),
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = note: expected `type_variance::Covariant<common::Lifetime<'_>>`
             found `type_variance::Covariant<common::Lifetime<'a>>`
  = note: but, the lifetime must be valid for the static lifetime...
note: ...so that the expression is assignable
 --> $DIR/fail_covariant.rs:6:38
  |
6 |       let _co: Co<Lifetime<'static>> = Co(
  |  ______________________________________^
7 | |         Covariant::<Lifetime<'a>>::default(),
8 | |     );
  | |_____^
  = note: expected `common::Co<common::Lifetime<'static>>`
             found `common::Co<common::Lifetime<'_>>`
